CPS-????/README.md

```diff
@@ -0,0 +1,242 @@
+  --tx-out "$(cat ${swapScriptAddrFile}) + 4000000 lovelace + 0 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.54657374546f6b656e0a + 250 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.4f74686572546f6b656e0a" \
+  --tx-out-inline-datum-file $swapDatumFile \
+  --withdrawal "$(cat ${stakingScriptAddrFile})+0" \
+  --withdrawal-script-file $stakingScriptFile \
+  --withdrawal-redeemer-file $swapRedeemerFile \
+  --tx-in-collateral bc54229f0755611ba14a2679774a7c7d394b0a476e59c609035e06244e1572bb#0 \
+  --change-address $(cat ../assets/wallets/01.addr) \
+  --protocol-params-file "${tmpDir}protocol.json" \
+  --testnet-magic 1 \
+  --out-file "${tmpDir}tx.body"
+```
+
+There are three new `--withdrawal` lines. The logic for the atomic swap is in the staking script. The actual spending script just checks that the required staking script was executed AND 0 ADA was withdrawn from the reward address.
+
+The only requirement to use this trick is that the swap address was created using the `s'` spending script and the `s''` staking script. The staking address DOES NOT need to actually have rewards or even be delegated for this trick to work.
+
+With this method, the main computation is only executed once.
+
+### The Drawbacks
+
+1. This trick now means that everything requires two plutus scripts instead of just one. While the spending script can be made extremely small, this doesn't change the fact that two are now needed.
+2. This trick DOES NOT stop the redundant executions; the spending script will still redundantly check if the required staking script was executed. This trick just minimizes the cost of the redundant executions by moving the heavy computation into a script that is only executed once per transaction.
+3. This is not the intended use for staking scripts.
+
+## Use Cases
+Any time a spending script's result depends on the entire transaction and not on any particular UTxO.
+
+1. P2P Atomic Swaps
+2. P2P Lending
+3. Certain DAO applications
+4. Certain Multisig applications that require more complex logic than what is possible with native scripts
+
+## Goals
+1. Stop redundantly executing spending scripts.
+2. Do not remove the option to allow script execution per UTxO. There are still cases where the script should validate based off individual UTxOs. Also, there could be cases where this option is more efficient than having to traverse the transaction's context to get the relevant datum. An example of this latter case is when only one UTxO is consumed from a script address.
+3. Try to use the ledger as is. The way scripts are stored onchain should not change unless absolutely necessary.
+4. Try to leave the developer experience as untouched as possible. We don't want to make developers re-learn plutus.
+5. Do not add too much new overhead to already resource constrained plutus scripts.
+
+## Open Questions
+1. Would this require a hardfork?
+2. Would this require a new version of plutus?
+
+---
+## Possible Solution
```

Comment:
I'd advise moving this into a new CIP document.

Reply:
I removed the proposal and will look to create a new CIP.